draft-ietf-ipsecme-ikev2-multiple-ke-00.xml

```diff
@@ -901,27 +904,39 @@ KEYMAT = prf+ (SK_d, SK(0) | Ni | Nr | SK(1) |  ... SK(n))
         <figure align="left"><artwork align="left"><![CDATA[
 Type     Description                   Used In                    
 -----------------------------------------------------------------
-<TBA>    Additional Key Exchange 1     (optional in IKE, AH, ESP)
-<TBA>    Additional Key Exchange 2     (optional in IKE, AH, ESP)
-<TBA>    Additional Key Exchange 3     (optional in IKE, AH, ESP)
-<TBA>    Additional Key Exchange 4     (optional in IKE, AH, ESP)
-<TBA>    Additional Key Exchange 5     (optional in IKE, AH, ESP)
-<TBA>    Additional Key Exchange 6     (optional in IKE, AH, ESP)
-<TBA>    Additional Key Exchange 7     (optional in IKE, AH, ESP)
+6        Additional Key Exchange 1     (optional in IKE, AH, ESP)
+7        Additional Key Exchange 2     (optional in IKE, AH, ESP)
+8        Additional Key Exchange 3     (optional in IKE, AH, ESP)
+9        Additional Key Exchange 4     (optional in IKE, AH, ESP)
+10       Additional Key Exchange 5     (optional in IKE, AH, ESP)
+11       Additional Key Exchange 6     (optional in IKE, AH, ESP)
+12       Additional Key Exchange 7     (optional in IKE, AH, ESP)
         ]]></artwork></figure>
 
         <t>This document defines a new Notify Message Type in the "Notify Message Types - Status Types" registry:</t>
 
         <figure align="center"><artwork align="left"><![CDATA[
-<TBA>       ADDITIONAL_KEY_EXCHANGE
+16441       ADDITIONAL_KEY_EXCHANGE
         ]]></artwork></figure>
 
         <t>and a new Notify Message Type in the "Notify Message Types - Error Types" registry:</t>
 
         <figure align="center"><artwork align="left"><![CDATA[
-<TBA>       STATE_NOT_FOUND
+47         STATE_NOT_FOUND
         ]]></artwork></figure>
 
+        <section title="Transform Type 4 Identifiers">
+            <t>While this document does not introduce any new Transform Type 4 Identifier, an additional
+            consideration should be made before adding an entry to the registry, that is the public value
+            of any new key exchange algorithm must fit into a single IKE_INTERMEDIATE message.  If there
+            is a need to add an algorithm that does not meet this restriction, considerations should be
+            made based on how the algorithm is used.  For example if the public value is distributed
+            out-of-band and no IKE_INTERMEDIATE exchange is required, then this restriction should not apply.
+            On the other hand, if the algorithm needs to use IKE_INTERMEDIATE exchanges, then there must
+            be documentation on how the public value is handled in the IKE_INTERMEDIATE exchange. 
+            </t>
+        </section>
```

Comment:
I would add a restriction that it is assumed that a KE method takes exactly one round-trip (one IKE_INTERMEDIATE exchange) and a shared key is available at the end of the exchange. 

If the KE method requires several round-trips, then additional considerations should be given in the document specifying such a method of how it is used in the context of this specification.